src/basic/calendarspec.c

```diff
@@ -581,7 +581,8 @@ static int calendarspec_from_time_t(CalendarSpec *c, time_t time) {
         CalendarComponent *year = NULL, *month = NULL, *day = NULL, *hour = NULL, *minute = NULL, *us = NULL;
         int r;
 
-        assert_se(gmtime_r(&time, &tm));
+        if (!gmtime_r(&time, &tm))
+                return -ERANGE;
```

Comment:
https://github.com/systemd/systemd/blob/master/src/test/test-calendarspec.c doesn't seem to cover this case. I think it would be useful to add it there too, because the test seems to be run even if `-Dslow_tests=true` is not used.